src/controllers/suggestions.js

```diff
@@ -0,0 +1,385 @@
+  const getAllForOpportunity = async (context) => {
+    const siteId = context.params?.siteId;
+    const opptyId = context.params?.opportunityId;
+
+    if (!hasText(siteId)) {
+      return badRequest('Site ID required');
+    }
+
+    if (!hasText(opptyId)) {
+      return badRequest('Opportunity ID required');
+    }
+
+    const suggestionEntities = await Suggestion.allByOpportunityId(opptyId);
+    // Check if the opportunity belongs to the site
+    if (suggestionEntities.length > 0) {
+      const oppty = await suggestionEntities[0].getOpportunity();
+      if (!oppty || oppty.getSiteId() !== siteId) {
+        return badRequest('Opportunity not found');
+      }
+    }
+    const suggestions = suggestionEntities.map((sugg) => SuggestionDto.toJSON(sugg));
+    return ok(suggestions);
+  };
+
+  /**
+   * Gets all suggestions for a given site, opportunity and status
+   * @param {Object} context of the request
+   * @returns {Promise<Response>} Array of suggestions response.
+   */
+  const getByStatus = async (context) => {
+    const siteId = context.params?.siteId;
+    const opptyId = context.params?.opportunityId;
+    const status = context.params?.status || undefined;
+    if (!hasText(siteId)) {
+      return badRequest('Site ID required');
+    }
+    if (!hasText(opptyId)) {
+      return badRequest('Opportunity ID required');
+    }
+    if (!hasText(status)) {
+      return badRequest('Status is required');
+    }
+
+    const suggestionEntities = await Suggestion.allByOpportunityIdAndStatus(opptyId, status);
+    // Check if the opportunity belongs to the site
+    if (suggestionEntities.length > 0) {
+      const oppty = await suggestionEntities[0].getOpportunity();
+      if (!oppty || oppty.getSiteId() !== siteId) {
+        return badRequest('Opportunity not found');
+      }
+    }
+    const suggestions = suggestionEntities.map((sugg) => SuggestionDto.toJSON(sugg));
+    return ok(suggestions);
+  };
+
+  /**
+   * Get a suggestion given a site, opportunity and suggestion ID
+   * @param {Object} context of the request
+   * @returns {Promise<Response>} Suggestion response.
+   */
+  const getByID = async (context) => {
+    const siteId = context.params?.siteId;
+    const opptyId = context.params?.opportunityId || undefined;
+    const suggestionId = context.params?.suggestionId || undefined;
+
+    if (!hasText(siteId)) {
+      return badRequest('Site ID required');
+    }
+
+    if (!hasText(opptyId)) {
+      return badRequest('Opportunity ID required');
+    }
+
+    if (!hasText(suggestionId)) {
+      return badRequest('Suggestion ID required');
+    }
+
+    const suggestion = await Suggestion.findById(suggestionId);
+    if (!suggestion || suggestion.getOpportunityId() !== opptyId) {
+      return notFound('Suggestion not found');
+    }
+    const opportunity = await suggestion.getOpportunity();
+    if (!opportunity || opportunity.getSiteId() !== siteId) {
+      return notFound('Suggestion not found');
```

Comment:
Opportunity not found?

Reply:
This API call is to get a suggestion by passing: siteID, opportunityID and suggestionID. IMO if any of the 3 parameters is wrong or there's no relation site -> oppty -> suggestion,  the error should be the same "Suggestion not found"

Reply:
Agreed to return unspecific "not found" message when the opportunity and the site don't match